tests/tests/conftest.py

```diff
@@ -0,0 +1,28 @@
+import os
+import pytest
+
+import Qemu
+
+@pytest.fixture(autouse=True)
+def run_before_and_after_tests(tmpdir):
+    """
+    Fixture to execute before and after a test is run
+    Even in case of test failure
+    """
+    # Setup: fill with any logic you want
+    # enable the debug flag if TDXTEST_DEBUG is set
+    debug = os.environ.get('TDXTEST_DEBUG', False)
+    if debug:
+        Qemu.QemuMachine.set_debug(debug)
+
+    yield # this is where the testing happens
+
+    # Teardown : fill with any logic you want
+
+def pytest_exception_interact(node, call, report):
+    """
+    Called at test failure
+    """
+    if report.failed:
+        # enable debug flag to avoid cleanup to happen
+        Qemu.QemuMachine.set_debug(True)
```

Comment:
why do we want to avoid cleanup when a test fails?  Don't we want the next text to start cleanly?  I understand if we running individually and we can set "debug", but not generally

Reply:
each qemu has its own working folder (/tmp/tdxtest-xxxxx/) that is not shared with other tests qemu instances